### PR TITLE
DAOS-5751 test: Remove deprecated control_method=API

### PR DESCRIPTION
**Pull request by berserk-fury**
In DAOS-5630 the DaosPool.create() method was removed from
pydaos. This patch updates the TestPool.create() method to
use the DmgCommand only for pool creation.

Signed-off-by: Jonathan Martinez Montes <jonathan.martinez.montes@intel.com>

### CODE REVIEW

**Review by shimizukko — APPROVED** [review]
LGTM

File: src/tests/ftest/pool/rebuild_with_io.py [hunk 1]
```diff
@@ -50,8 +50,7 @@ def test_rebuild_with_io(self):
         :avocado: tags=all,pool,rebuild,pr,medium,rebuildwithio
         """
         # Get the test params
-        pool = TestPool(self.context, self.log,
-                        dmg_command=self.get_dmg_command())
+        pool = TestPool(self.context, self.get_dmg_command(), log=self.log)
```
Comment (phender) [inline src/tests/ftest/pool/rebuild_with_io.py:53]: The `log` argument is defunct for TestPool.create():
```suggestion
        pool = TestPool(self.context, self.get_dmg_command())
```

File: src/tests/ftest/pool/rebuild_tests.py [hunk 2]
```diff
@@ -45,8 +45,8 @@ def run_rebuild_test(self, pool_quantity):
         pools = []
         containers = []
         for index in range(pool_quantity):
-            pools.append(TestPool(self.context, self.log,
-                                  dmg_command=self.get_dmg_command()))
+            pools.append(TestPool(self.context, self.get_dmg_command(),
+                                  log=self.log))
```
Comment (phender) [inline src/tests/ftest/pool/rebuild_tests.py:49]: The `log` argument is defunct for TestPool.create():
```suggestion
            pools.append(TestPool(self.context, self.get_dmg_command()))
```

File: src/tests/ftest/pool/rebuild_no_cap.py [hunk 3]
```diff
@@ -48,8 +48,8 @@ def test_rebuild_no_capacity(self):
         :avocado: tags=all,medium,pr,pool,rebuild,nocap
         """
         # Get the test params
-        self.pool = TestPool(self.context, self.log,
-                             dmg_command=self.get_dmg_command())
+        self.pool = TestPool(self.context, self.get_dmg_command(),
+                             log=self.log)
```
Comment (phender) [inline src/tests/ftest/pool/rebuild_no_cap.py:52]: The `log` argument is defunct for TestPool.create():
```suggestion
        self.pool = TestPool(self.context, self.get_dmg_command())
```

File: src/tests/ftest/pool/info_tests.py [hunk 4]
```diff
@@ -47,8 +47,8 @@ def test_pool_info_query(self):
         :avocado: tags=all,tiny,pr,pool,smoke,infotest
         """
         # Get the test params
-        self.pool = TestPool(self.context, self.log,
-                             dmg_command=self.get_dmg_command())
+        self.pool = TestPool(self.context, self.get_dmg_command(),
+                             log=self.log)
```
Comment (phender) [inline src/tests/ftest/pool/info_tests.py:51]: The `log` argument is defunct for TestPool.create():
```suggestion
        self.pool = TestPool(self.context, self.get_dmg_command())
```

File: src/tests/ftest/pool/evict_test.py [hunk 5]
```diff
@@ -50,8 +50,7 @@ def connected_pool(self, hostlist, targets=None):
             TestPool (object)
 
         """
-        pool = TestPool(self.context, self.log,
-                        dmg_command=self.get_dmg_command())
+        pool = TestPool(self.context, self.get_dmg_command(), log=self.log)
```
Comment (phender) [inline src/tests/ftest/pool/evict_test.py:53]: The `log` argument is defunct for TestPool.create():
```suggestion
        pool = TestPool(self.context, self.get_dmg_command())
```

File: src/tests/ftest/pool/destroy_rebuild.py [hunk 6]
```diff
@@ -48,8 +48,7 @@ def test_destroy_while_rebuilding(self):
         :avocado: tags=all,pr,medium,pool,destroypoolrebuild
         """
         # Get the test parameters
-        self.pool = TestPool(self.context, self.log,
-                             dmg_command=self.get_dmg_command())
+        self.pool = TestPool(self.context, self.get_dmg_command(), log=self.log)
```
Comment (phender) [inline src/tests/ftest/pool/destroy_rebuild.py:51]: The `log` argument is defunct for TestPool.create():
```suggestion
        self.pool = TestPool(self.context, self.get_dmg_command())
```

**Review by phender — CHANGES_REQUESTED** [review]

**Review by cibervicho — DISMISSED** [review]

**Comment by sylviachanoiyee** [comment]
@berserk-fury , can you leave a comment in this PR with JIRA tickets to the failures in Jenkins? thanks!!

**Pull request merged**